### PR TITLE
Add necessary flags for naïve OS X support

### PR DESCRIPTION
**Pull request by neonichu**
We need to at least set `-target`, `-sdk` and pass `-all_load` to make it work.

### CODE REVIEW

File: Sources/main.swift [hunk 1]
```diff
@@ -1,11 +1,24 @@
 import Commander
 import PathKit
 
+private func target_specific_arguments() throws -> [String] {
+#if os(OSX)
+  return [
+    "-sdk",
+    "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk",
+    "-target", "x86_64-apple-macosx10.10",
```
Comment (kylef) [inline Sources/main.swift]: What if this isn't true?

Comment (neonichu) [inline Sources/main.swift]: :boom: — that's why it's naïve 😛

Comment (neonichu) [inline Sources/main.swift]: Might be nice to extract the options by running `swift build` on an empty thing?

Comment (kylef) [inline Sources/main.swift]: SPM is going to get test support sometime. So perhaps it isn't worth investing time in it.

Will merge this for the interim.

**Pull request merged**